src/commands/Pokemon/pokedex.ts

```diff
@@ -0,0 +1,176 @@
+import { toTitleCase } from '@klasa/utils';
+import { MessageEmbed } from 'discord.js';
+import { CommandStore, KlasaMessage } from 'klasa';
+import { SkyraCommand } from '../../lib/structures/SkyraCommand';
+import { getPokemonDetailsByFuzzy, GraphQLPokemonResponse, POKEMON_GRAPHQL_API_URL, POKEMON_EMBED_THUMBNAIL, parseBulbapediaURL } from '../../lib/util/Pokedex';
+import { fetch } from '../../lib/util/util';
+
+const BaseStats: Record<string, string> = {
+	hp: 'HP',
+	attack: 'ATK',
+	defense: 'DEF',
+	specialattack: 'SPA',
+	specialdefense: 'SPD',
+	speed: 'SPE'
+};
+
+export default class extends SkyraCommand {
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			aliases: ['pokemon', 'dex'],
+			cooldown: 10,
+			description: language => language.tget('COMMAND_POKEDEX_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_POKEDEX_EXTENDED'),
+			requiredPermissions: ['EMBED_LINKS'],
+			usage: '<pokemon:str>'
+		});
+	}
+
+	public async run(message: KlasaMessage, [pokemon]: [string]) {
+		try {
+			const embedTranslations = message.language.tget('COMMAND_POKEDEX_EMBED_DATA');
+			const { getPokemonDetailsByFuzzy: poke } = (await this.fetchAPI(message, pokemon)).data;
+
+			const baseStats: string[] = [];
+			let abilities = '';
+			let genders = '';
+			let evoChain = `**${toTitleCase(poke.species)}**`;
+
+			// Parse abilities
+			for (const [type, ability] of Object.entries(poke.abilities)) {
+				if (ability !== null) {
+					switch (type) {
+						case 'first':
+							abilities += ability;
+							break;
+						case 'hidden':
+							abilities += `, *${ability}*`;
+							break;
+						case 'second':
+						case 'special':
+							abilities += `, ${ability}`;
+					}
+				}
+			}
+
+			// Parse base stats
+			for (const [stat, value] of Object.entries(poke.baseStats)) {
+				baseStats.push(`${BaseStats[stat]}: **${value}**`);
+			}
+
+			// Parse gender ratio
+			for (const [gender, ratio] of Object.entries(poke.gender)) {
+				switch (gender) {
+					case 'male':
+						genders += `${ratio} ♂ `;
+						break;
+					case 'female':
+						genders += ` | ${ratio} ♀`;
+						break;
+				}
+			}
+
+			// Set evochain if there are no evolutions
+			if (!poke.evolutions?.length && !poke.preevolutions?.length) {
+				evoChain += ' (No Evolutions)';
+			}
+
+			// Parse pre-evolutions and add to evochain
+			if (poke.preevolutions?.length) {
+				const { evolutionLevel } = poke.preevolutions[0];
+				evoChain = `\`${toTitleCase(poke.preevolutions[0].species)}\` ${evolutionLevel ? `(${evolutionLevel})` : ''} → ${evoChain} **(${poke.evolutionLevel})**`;
+
+				// If the direct pre-evolution has another pre-evolution (charizard -> charmeleon -> charmander)
+				if (poke.preevolutions[0].preevolutions?.length) {
+					evoChain = `\`${toTitleCase(poke.preevolutions[0].preevolutions[0].species)}\` → ${evoChain}`;
+				}
+			}
```

Comment:
This can be refactored, preferably with a helper function.

Essentially, you can do a single `evoChain` assignment instead of 2 (3 with the initial set).